File: src/main/java/io/github/oxi1224/remotecontroller/server/Opcode.java
package io.github.oxi1224.remotecontroller.server;

public enum Opcode {
  CLOSE(0x0),
  CONNECT(0x1),

  COMMAND(0x6),
  RESULT(0x7),

  PING(0xD),
  PONG(0xE),
  UNUSED(0xF);

  public static Opcode findByValue(int value) {
    for (Opcode c : values()) {
      if (c.getValue() == value) return c;
    }
    return Opcode.UNUSED;
  }
  private final byte code;
  Opcode(int code) { this.code = (byte)code; }
  public byte getValue() { return this.code; }
}

File: client/src/opcode.cpp
#include "opcode.hpp"

Opcode toOpcode(byte val) {
  switch (val) {
    case CLOSE:
    case CONNECT:
    case COMMAND:
    case RESULT:
    case PING:
    case PONG:
    case UNUSED:
      return static_cast<Opcode>(val);
    default:
      return UNUSED;
  }
}

bool opcode_equals(Opcode opcode, byte value) {
  return opcode == toOpcode(value);
}

